### DISCUSSION TITLE
OpenAI sampling parameter "seed"

### DISCUSSION THREAD

**Discussion by Desmond819** [sgl-project/sglang#683] [Q&A] [answered]
Is seed used in the sampling?

**Comment by Desmond819** [comment]
does anyone know? How can I configure the server to accept the seed parameter?
<img width="634" alt="image" src="https://github.com/user-attachments/assets/a6f01dbb-976c-408a-8c22-2b3c01eaa203">

**Comment by hrukalive** [comment, marked as answer]
This is not a definite answer, I am just roaming and reading codes. I found the following definitions, and it seems like no `seed` parameter involved.
https://github.com/sgl-project/sglang/blob/001b0bdd089a626ada2ee217fdc59b4212f0b461/python/sglang/srt/managers/io_struct.py#L30-L51
https://github.com/sgl-project/sglang/blob/001b0bdd089a626ada2ee217fdc59b4212f0b461/python/sglang/srt/sampling_params.py#L23-L38

When adapting OpenAI requests, the `seed` is dropped.
https://github.com/sgl-project/sglang/blob/001b0bdd089a626ada2ee217fdc59b4212f0b461/python/sglang/srt/openai_api/protocol.py#L130-L153
https://github.com/sgl-project/sglang/blob/001b0bdd089a626ada2ee217fdc59b4212f0b461/python/sglang/srt/openai_api/adapter.py#L352-L364

I take this as sglang server does not take seed parameter yet.

**Comment by Desmond819** [comment]
Thanks for your reply. Does it mean I just need to add them back to these 2 code to make it work?

**Comment by Ying1123** [comment]
seed parameter has not been supported. May I ask your specific use case?

**Comment by Desmond819** [comment]
I need to provide a deterministic result compared to the groundtruth response which a random seed will be given when it's generated.

**Comment by Desmond819** [comment]
After add the seed parameter to the adaptor. I also updated tp_worker.py by moving the set_random_seed function from initialization to handle_generate_request(). Not sure if this is the right way.

<img width="799" alt="image" src="https://github.com/user-attachments/assets/36acf51e-653c-442d-9ef8-4a86cb9928f9">
